src/VCR/Util/CurlHelper.php

```diff
@@ -175,24 +175,39 @@ public static function setCurlOptionOnRequest(Request $request, $option, $value,
             case CURLOPT_HEADER:
             case CURLOPT_WRITEFUNCTION:
             case CURLOPT_HEADERFUNCTION:
+            case CURLOPT_UPLOAD:
                 // Ignore header, file and writer functions.
                 // These options are stored and will be handled later in handleOutput().
                 break;
-            case CURLOPT_READFUNCTION:
-                // Guzzle provides a callback to let curl read the body string.
-                // To get the body, this callback is called manually.
-                if (is_null($value)) {
-                    $request->setCurlOption($option, $value);
-                    break;
-                }
-                $bodySize = $request->getCurlOption(CURLOPT_INFILESIZE);
-                Assertion::notEmpty($bodySize, "To set a CURLOPT_READFUNCTION, CURLOPT_INFILESIZE must be set.");
-                $body = call_user_func_array($value, array($curlHandle, fopen('php://memory', 'r'), $bodySize));
-                $request->setBody($body);
-                break;
             default:
                 $request->setCurlOption($option, $value);
                 break;
         }
     }
+    
+    /**
+     * Makes sure we've properly handled the POST body, such as ensuring that
+     * CURLOPT_INFILESIZE is set if CURLOPT_READFUNCTION is set.
+     *
+     * @param Request  $request Request to set cURL option to.
+     * @param resource $curlHandle cURL handle associated with the request.
+     */
+    public static function validateCurlPOSTBody(Request $request, $curlHandle = null)
+    {
+        $readFunction = $request->getCurlOption(CURLOPT_READFUNCTION);
+        if (is_null($readFunction)) {
+            return;
+        }
+        
+        // Guzzle 4 sometimes sets the post body in CURLOPT_POSTFIELDS even if
```

Comment:
You linked to the current version which is Guzzle 5. The case I was handling occurs in [Guzzle 4](https://github.com/guzzle/guzzle/blob/4.2.3/src/Adapter/Curl/CurlFactory.php). In `getDefaultOptions` you can see that `CURLOPT_READFUNCTION` is set by default while in `applyBody` you can have `CURLOPT_POSTFIELDS` set instead of `CURLOPT_UPLOAD` and `CURLOPT_INFILESIZE`.